core/calculator.js

```diff
@@ -670,12 +834,28 @@ function generate_chart(edges, node_quantities) {
 	var nodes = {};
 	for (let column_id in columns) {
 		for (let node_id in columns[column_id]) {
-			var node_name = columns[column_id][node_id];
+			let node_name = columns[column_id][node_id];
+
+			let input = get_input_size(edges, node_name);
+			let output;
+			if (node_name.endsWith(inventory_label_suffix)) {
+				output = used_from_inventory[node_name.replace(inventory_label_suffix, "")];
```

Comment:
I would have to follow through the code path again to check this myself but maybe you know as it is fresher in your mind. Will there ever be a case where `used_from_inventory[node_name.replace(inventory_label_suffix."")]` `!=` `node_quantities[node_name]`?

Reply:
That is a very good question, that I will look into as soon, as I get back to my PC.

Reply:
It is always the case, if an item is needed and it will not be sourced from the inventory:
![image](https://user-images.githubusercontent.com/11810519/119227415-02739e80-bb0e-11eb-8946-1b30157ce021.png)

Reply:
from your example it looks to me like they are equal in this case
`used_from_inventory["Chest"] = node_quantities["Chest [from Inventory]"] = 59`
`used_from_inventory["Hopper"] = node_quantities["Hopper [from Inventory]"] = 5`
`used_from_inventory["Redstone"] = node_quantities["Redstone [from Inventory]"] = 16`

so you should be able to remove that first if statement so
```
			if (node_name.endsWith(inventory_label_suffix)) {
				output = used_from_inventory[node_name.replace(inventory_label_suffix, "")];
			}
			else if (node_quantities[node_name] !== undefined) {
				output = node_quantities[node_name];
			}
			else {
				output = 0;
			}
```
becomes just
```
			if (node_quantities[node_name] !== undefined) {
				output = node_quantities[node_name];
			}
			else {
				output = 0;
			}
```

Reply:
Ah, I see what you mean. I will look into that, just to check for edge cases.

Reply:
You were totally right about that one. Comitted & pushed your suggestion.